vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/HasThemeVariant.java

```diff
@@ -0,0 +1,40 @@
+package com.vaadin.flow.component;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Mixin interface that allows adding and removing typed theme variants to /
+ * from a component
+ *
+ * @param <TVariantEnum>
+ *            The specific theme variant enum type
+ */
+public interface HasThemeVariant<TVariantEnum extends ThemeVariant>
+        extends HasTheme {
+    /**
+     * Adds theme variants to the component.
+     *
+     * @param variants
+     *            theme variants to add
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
Got a possible heap pollution warning here, which results from using var args with a generic type. I should look into this some more, but maybe someone knows what the impact is.

Reply:
To my knowledge it's a compiler warning that points out that`T<U>[]` (i.e., `T<U>...`) is erased to `T...` and has the compile-time supertype of`Object[]`. That can can lead to unexpected runtime type errors that are not picked up as errors by the compiler. `java.lang.SafeVarargs` has a decent explanation of the issue (and will also silence the warning).